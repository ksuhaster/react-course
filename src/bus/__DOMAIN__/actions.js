// Types
import { types } from './types';

export const domainActions = Object.freeze({
    domainAction: () => {
        return {
            type: types.DOMAIN_ACTION
        }
    }
});