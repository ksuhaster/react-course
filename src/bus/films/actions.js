// Types
import { types } from './types';

import { api } from '../../api';

export const filmsActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILMS_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILMS_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILMS_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILMS_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.FILMS_FETCH_ASYNC
        });

        dispatch(filmsActions.startFetching());

        const response = await api.films.fetch();

        if (response.status === 200) {
            const { results } = await response.json();
            dispatch(filmsActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(filmsActions.setFetchingError(error));
        }

        dispatch(filmsActions.stopFetching());
    }
});


export const filmActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILM_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILM_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILM_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILM_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    fetchAsync: (id) => async (dispatch) => {
        dispatch({
            type: types.FILM_FETCH_ASYNC
        });

        dispatch(filmActions.startFetching());

        const response = await api.film.fetch(id);

        if (response.status === 200) {
            const results = await response.json();
            dispatch(filmActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(filmActions.setFetchingError(error));
        }

        dispatch(filmActions.stopFetching());
    }
});