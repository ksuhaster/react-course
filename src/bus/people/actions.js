// Types
import { types } from './types';

import { api } from '../../api';

export const peopleActions = Object.freeze({
    // Sync
    fill: (payload) => {
        return {
            type: types.PEOPLE_FILL,
            payload
        }
    },
    // Async
    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.PEOPLE_FETCH_ASYNC
        });

        const response = await api.people.fetch();
        const { results } = await response.json();

        dispatch(peopleActions.fill(results));
    }
});