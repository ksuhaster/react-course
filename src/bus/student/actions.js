// Types
import { types } from './types';

export const studentActions = Object.freeze({
    setStudent: (payload) => {
        return {
            type: types.SET_NAME,
            payload
        }
    }
});