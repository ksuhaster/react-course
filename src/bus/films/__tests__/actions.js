import { filmsActions, filmActions} from "../actions";
import expectExport from "expect";

describe("film list actions:", () => {
    // Sync
    test("startFetching", () => {
        expectExport(filmsActions.startFetching()).toMatchSnapshot();
    });
    test("stopFetching", () => {
        expectExport(filmsActions.stopFetching()).toMatchInlineSnapshot(`
      Object {
        "type": "FILMS_STOP_FETCHING",
      }
    `);
    });
    test("fill", () => {
        expectExport(filmsActions.fill()).toMatchSnapshot();
    });
    test("setFetchingError", () => {
        expectExport(filmsActions.fill()).toMatchSnapshot();
    });

    // Async
    test("fetchAsync", () => {
        expectExport(filmsActions.fetchAsync()).toMatchSnapshot();
    });
});


describe("one film actions:", () => {
    // Sync
    test("startFetching", () => {
        expectExport(filmActions.startFetching()).toMatchSnapshot();
    });
    test("stopFetching", () => {
        expectExport(filmActions.stopFetching()).toMatchInlineSnapshot(`
      Object {
        "type": "FILM_STOP_FETCHING",
      }
    `);
    });
    test("fill", () => {
        expectExport(filmActions.fill()).toMatchSnapshot();
    });
    test("setFetchingError", () => {
        expectExport(filmActions.fill()).toMatchSnapshot();
    });

    // Async
    test("fetchAsync", () => {
        expectExport(filmActions.fetchAsync()).toMatchSnapshot();
    });
});
