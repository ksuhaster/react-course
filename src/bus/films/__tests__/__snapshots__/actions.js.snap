// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`film list actions: fetchAsync 1`] = `[Function]`;

exports[`film list actions: fill 1`] = `
Object {
  "payload": undefined,
  "type": "FILMS_FILL",
}
`;

exports[`film list actions: setFetchingError 1`] = `
Object {
  "payload": undefined,
  "type": "FILMS_FILL",
}
`;

exports[`film list actions: startFetching 1`] = `
Object {
  "type": "FILMS_START_FETCHING",
}
`;

exports[`one film actions: fetchAsync 1`] = `[Function]`;

exports[`one film actions: fill 1`] = `
Object {
  "payload": undefined,
  "type": "FILM_FILL",
}
`;

exports[`one film actions: setFetchingError 1`] = `
Object {
  "payload": undefined,
  "type": "FILM_FILL",
}
`;

exports[`one film actions: startFetching 1`] = `
Object {
  "type": "FILM_START_FETCHING",
}
`;
